stablehlo/dialect/TypeInference.cpp

```diff
@@ -376,8 +376,8 @@ verifyWindowAttributesAndInferWindowDimensions(
 //            padding_high,
 //                         dilatedBound(window_shape[d]))
 //      where (padding_low, padding_high) is the padding-pair for d.
-SmallVector<int64_t> inferWindowOutputShape(
-    const ArrayRef<int64_t> baseShape, const ArrayRef<WindowDimension> window) {
+SmallVector<int64_t> inferWindowOutputShape(ArrayRef<int64_t> baseShape,
```

Comment:
Looks like this issue `const ArrayRef` has existed for awhile, any reason we are just catching this now? Any preventative steps we should take?

Reply:
I came across this issue after receiving a ClangTidy warning during the integrate. I think that happened because the integrate touched TypeInference.{cpp,h}, so the presubmits decided to check those files with ClangTidy.

One way to avoid these issues in the future is to codify fixing ClangTidy warnings during integrates (I don't think this is a clear-cut part of the integrate process at the moment). Another way is to run ClangTidy as part of GitHub actions, maybe nightly or something if this ends up being too slow.